### PR TITLE
testing/deployer: rename various terms to better align with v2 and avoid confusion

### PR DESCRIPTION
**Pull request by rboyer**
### Description

Conceptually renaming the following topology terms to avoid confusion with v2 and to better align with it:

- ServiceID -> ID
  - it's a generic namespace-scoped identifier
- Service -> Workload
  - Most places it implies "instance of a service" except for a few stray ones where the original nomenclature remains in place.
- Upstream -> Destination

Methods, functions, and types using the old names were retained as shims to avoid having to babysit the ce/enterprise merge, but the use of the deprecated words will be removed in follow up commits to each repository.

### CODE REVIEW

File: testing/deployer/topology/topology.go [hunk 1]
```diff
@@ -410,26 +410,18 @@ func (c *Cluster) SortedNodes() []*Node {
 	return out
 }
 
-func (c *Cluster) FindService(id NodeServiceID) *Service {
```
Comment (rboyer) [inline testing/deployer/topology/topology.go:413]: This method was completely unused.

File: testing/deployer/topology/ids.go [hunk 2]
```diff
@@ -9,41 +9,6 @@ import (
 	"github.com/hashicorp/consul/api"
 )
 
-type NodeServiceID struct {
```
Comment (rboyer) [inline testing/deployer/topology/ids.go:12]: This type was completely unused.

File: testing/deployer/topology/topology.go [hunk 3]
```diff
@@ -705,26 +699,21 @@ func (n *Node) DigestExposedPorts(ports map[int]int) bool {
...
+	id.Normalize()
+	for _, wrk := range n.Workloads {
+		if wrk.ID == id {
+			return wrk
 		}
 	}
-	panic("service not found: " + sid.String())
-}
-
-type ServiceAndNode struct {
```
Comment (rboyer) [inline testing/deployer/topology/topology.go:725]: unused type

**Review by ndhanushkodi — APPROVED** [review]
🎉

**Pull request merged**